packages/server/src/oauth/token.ts

```diff
@@ -550,14 +550,24 @@ async function validateClientIdAndSecret(
     return false;
   }
 
+  let failed = false;
+
   // Use a timing-safe-equal here so that we don't expose timing information which could be
   // used to infer the secret value
   if (!timingSafeEqualStr(client.secret, clientSecret)) {
+    failed = true;
+  }
+  const secondarySecret = client.retiringSecret ?? client.secret;
```

Comment:
Won't this basically make us do the same comparison twice in the case where there is no retiring secret?

Reply:
Or is this intentional so that this endpoint can't be used to infer the existence of a retiring secret? 🤔

Reply:
Yep, intentional to minimize information leakage

Reply:
Cool, can you add a comment about that for future reference?